### PR TITLE
Format and update locales

### PR DESCRIPTION
**Pull request by afh**
This PR draft adds the `script/update-locales.sh` script which will format the translations files, sorting them alphabetically by key and add any missing translations from `en_US.json` to the language specific translations files prefixing the value with `_TSTR_` to indicate a missing translation.

This should allow to more easily track which languages need translation for which keys.

Issuing this a draft to get feedback on whether this is headed into the right direction.

**TODO:**
- [ ] Figure out a way to detect translation key renames or obsoleted or misspelled keys in non-source (source being `en_US.json`) files.

### CODE REVIEW

File: script/update-locales.sh [hunk 1]
```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+# Format and update translations from a source file (en_US.json by default)
+#
+# Requires jq (https://stedolan.github.io/jq/) and sponge
+# (https://www.putorius.net/linux-sponge-soak-up-standard-input-and-write-to-a-file.html)
```
Comment (afh) [inline script/update-locales.sh]: @clangen Do you find introducing these utilities as a developer dependency acceptable?

**Comment by afh** [comment]
@clangen What is the process currently to get new strings translated into the various languages that `musikcube` supports?

Looking at the state of the languages (see below) it seems that French 🇫🇷, Italian 🇮🇹, and Japanese 🇯🇵 would benefit from a closer look from someone fluent in the respective language.

I could look into 🇮🇹 and 🇫🇷, but only have beginner level knowledge of the two.


```
% grep -c _TSTR_ src/musikcube/data/locales/*.json
src/musikcube/data/locales/de_DE.json:0
src/musikcube/data/locales/en_US.json:0
src/musikcube/data/locales/es_ES.json:0
src/musikcube/data/locales/fr_FR.json:163
src/musikcube/data/locales/it_IT.json:99
src/musikcube/data/locales/ja_JP.json:32
src/musikcube/data/locales/ru_RU.json:1
src/musikcube/data/locales/zh_CN.json:0
```

**Comment by clangen** [comment]
Again, apologies for the delay reviewing this change, I know it's probably quite annoying. I've been busy lately, and have been using my limited time to take care of a couple large dependency updates that have been sort of rotting away over the past couple years (specifically: Android 12, 13 support with improved media handling and upgrades to PDCurses, the curses library we use for Windows builds).

> @clangen Do you find introducing these utilities as a developer dependency acceptable?

Totally fine by me as long as the tools are generally easy to obtain via package managers and not overly obscure. No issues with `jq`, specifically.

> @clangen What is the process currently to get new strings translated into the various languages that musikcube supports?

It's very ad-hoc. We don't have dedicated translators, so people just help out whenever they can, if they have time. 

I'm cool to merge this change as soon as the conflict is sorted out. :)

**Comment by afh** [comment]
Thanks for taking the time to review this PR, @clangen, very much appreciated, and thank you for providing more context on where your attention has been lately. Were you able to finish the large dependency updates?

When it comes to PRs from me, I'm patient and will send a friendly reminder every now and then :)

I went ahead and replace the use of `sponge(1)` with file redirection `>` and `mv`. While sponge is a handy tool it is rather uncommon and I'm uncertain how widely the moreutils, which provides sponge, is available in different environments.

**Comment by clangen** [comment]
> Thanks for taking the time to review this PR, @clangen, very much appreciated, and thank you for providing more context on where your attention has been lately. Were you able to finish the large dependency updates?

Yep! Only I have tested them so far, but the Android app is back up and running smoothly with modernized playback notification handling, support for Android 13 and various bug other fixes. `PDCurses` changes have been merged with a bit of custom sauce to make things more stable, and I'm currently working with the upstream maintainer to get a couple fixes merged. Next up is to take a look at the raw `Pipewire` output plugin for Linux, as I've seen a couple reports of problems with it lately. :)

**Comment by clangen** [comment]
Thanks for the update, merged!

**Comment by afh** [comment]
That sounds quite productive and successful, well done, @clangen!

Thanks for merging this PR :)

**Pull request merged**